### ISSUE TITLE
bug: Cognito access token - username is sub, not email

### ISSUE THREAD

**Issue by sam-super** [localstack/localstack#11442]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

If you setup the pool using `--username-attributes 'email'` then the subsequent access token for any user will have the users `sub` as the `username` field in the jwt, rather than the email address

### Expected Behavior

the `username` in the access token should be the `email`, not the `sub`

### How are you starting LocalStack?

With a docker-compose file

### Steps To Reproduce

Have localstack-pro 3.7 running (3.6 doesn't have the same problem):

Run this in a bash script and note the last output `username` property:

```#!/usr/bin/env bash

set -e

shopt -s expand_aliases
alias aws="aws --region=us-east-1 --endpoint-url=http://localhost:4566"

function jwt_decode(){
    jq -R 'split(".") | .[1] | @base64d | fromjson' <<< "$1"
}

unset AWS_PROFILE

TS=$(date +%s)
USER_POOL_NAME="example_user_pool_$TS"
USER_POOL_CLIENT_NAME="example_user_pool_client$TS"
USER_EMAIL="test@example.com"
USER_PASSWORD="P@s5W0rdddd"

aws cognito-idp create-user-pool \
--pool-name $USER_POOL_NAME \
--username-attributes 'email'

# THE ISSUE IS ABOVE
# - setting "--username-attributes 'email'" means the AccessToken's 'username' will be the sub (not the email address)

USER_POOL_ID=$(aws cognito-idp list-user-pools --max-results 100 \
--query "UserPools[?Name==\`$USER_POOL_NAME\`].Id" --output text)

aws cognito-idp create-user-pool-client \
    --user-pool-id $USER_POOL_ID \
    --client-name $USER_POOL_CLIENT_NAME

USER_POOL_CLIENT_ID=$(aws cognito-idp list-user-pool-clients \
    --user-pool-id $USER_POOL_ID \
    --max-results 10 \
    --query "UserPoolClients[?ClientName==\`$USER_POOL_CLIENT_NAME\`].ClientId" --output text)

aws cognito-idp admin-create-user \
    --user-pool-id $USER_POOL_ID \
    --username $USER_EMAIL \
    --temporary-password $USER_PASSWORD

aws cognito-idp admin-set-user-password \
    --user-pool-id $USER_POOL_ID \
    --username $USER_EMAIL \
    --password $USER_PASSWORD \
    --permanent

ACCESS_TOKEN=$(aws cognito-idp initiate-auth \
    --auth-flow USER_PASSWORD_AUTH \
    --client-id $USER_POOL_CLIENT_ID \
    --auth-parameters USERNAME=$USER_EMAIL,PASSWORD=$USER_PASSWORD \
    --query 'AuthenticationResult.AccessToken' --output text
    )

DECODED=$(jwt_decode $ACCESS_TOKEN)

echo $DECODED;
```
e.g. `"username": "9b6ba93b-6d5e-404b-9488-5fad535995e2"`
```
{ "exp": 1725281937, "iss": "http://localhost:4566/us-east-1_2c539f62e21b45218fb27643a2c9ac8a", "sub": "9b6ba93b-6d5e-404b-9488-5fad535995e2", "auth_time": 1725278337, "iat": 1725278337, "event_id": "b2975578-f5e0-451f-93ec-81444958fab3", "token_use": "access", "username": "9b6ba93b-6d5e-404b-9488-5fad535995e2", "jti": "cffc18d5-2de3-468c-9978-c69da5ffdd93", "client_id": "zfhpyjhpoqc7995jr72fdm5od9", "scope": "aws.cognito.signin.user.admin" }
```
running against `localstack/localstack-pro:3.6` or removing `--username-attributes 'email'` fix the issue (the username goes back to being the email).

### Environment

```markdown
- OS: macos
- LocalStack: 
  LocalStack version: 3.7
```

**Comment by bentsku** [comment]
Hello @sam-super and thanks for your report!

I've just tried your steps against AWS, and already from the response in `admin-create-user`, the `Username` field is the `sub` field.

When running your code against AWS, it returns the `sub` as the `username` in the access token. 

Also, the AWS documentation specifies that if you're using `UsernameAttributes` in your user pool, the `Username` will be equivalent to the `sub`: https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#heading:r2o:

> The `SignUp` API generates a persistent UUID for your user, and uses it internally as the immutable username attribute. This UUID has the same value as the sub claim in the user identity token.

Could you please confirm that if you run the sample you've provided against AWS, you also get the `sub` field as your username? 

You also mention that removing `UsernameAttributes` when creating your user pool makes the username go back to the `email`. This is also the behavior in AWS.

**Comment by sam-super** [comment]
hi @bentsku , thanks for looking into it! I checked and can confirm our aws tokens have the sub in the username too. Sorry i didn't check that myself first. Looks like it's a case of us asserting the incorrect behaviour in our tests <=3.6. 

So looks like the localstack team actually fixed this bug in cognito in 3.7 👏 . nice one!

**Comment by bentsku** [comment]
Awesome, thanks for coming back to us! Thanks for the feedback 🙏 and no worries for no checking earlier, it's on us for the previous behavior. Don't hesitate if there's anything else!
